src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,175 @@
+                isError: true,
+                message: errorThrown
+            });
+        });;
+    } else {
+        storyPromise.reject({
+            isError: true,
+            message: "No story ID provided"
+        });
+    }
+
+    return storyPromise;
+};
+
+/* Loads a story Browse page and populates it with a set of stories
+ * - "theme": the theme of the story Browse page (pass "page" in for base theme)
+ * - "options": additional options to merge into the Browse page
+ */
+sjrk.storyTelling.loadBrowse = function (theme, options) {
+    var storiesPromise = fluid.promise();
+
+    $.get("/stories", function (data) {
+        var browseResponse = JSON.parse(data);
+
+        options = options || {};
+        options.distributeOptions = {
+            "target": "{that storyBrowser}.options.model",
+            "record": browseResponse
+        };
+
+        var storyBrowseComponent;
+        if (theme === "base") {
+            storyBrowseComponent = sjrk.storyTelling.page.storyBrowse(options);
+        } else {
+            storyBrowseComponent = sjrk.storyTelling[theme].storyBrowse(options);
+        }
+
+        storiesPromise.resolve(storyBrowseComponent);
+    }).fail(function (jqXHR, textStatus, errorThrown) {
+        storiesPromise.reject({
+            isError: true,
+            message: errorThrown
+        });
+    });
+
+    return storiesPromise;
+};
+
+/* Loads custom theme files if and only if one of either the theme override or
+ * server configuration theme are present AND not set to the base theme.
+ * Once complete, the provided callback function is called and the theme name is
+ * passed into it. If custom theme files are not loaded, the "base" theme is passed.
+ * - "callback": a function to call once everything has completed. Will be called
+ *               regardless of whether theme information was specified or retrieved
+ * - "themeOverride": allows overriding of the theme stored in the configuration
+ */
+sjrk.storyTelling.loadThemedPage = function (callback, themeOverride) {
+    var loadPromise = fluid.promise();
+
+    var callbackFunction = typeof callback === "function" ? callback : fluid.getGlobalValue(callback);
+
+    $.get("/clientConfig").then(function (data) {
+        var theme = themeOverride ? themeOverride : data.clientConfig.theme;
+
+        if (theme && theme !== "base") {
+            return sjrk.storyTelling.loadCustomThemeFiles(callbackFunction, theme).then(function () {
+                loadPromise.resolve(theme);
+            }, function (jqXHR, textStatus, errorThrown) {
+                loadPromise.reject({
+                    isError: true,
+                    message: errorThrown
+                });
+            });
+        } else {
+            theme = "base";
+            callbackFunction(theme);
+            loadPromise.resolve(theme);
+        }
+    }, function (jqXHR, textStatus, errorThrown) {
+        callbackFunction("base"); //default to the base theme page
```

Comment:
replace "base" with the variable `theme` in case there's a name change for the base theme in the future. Please avoid hardcoded strings whenever possible. Thanks.

Reply:
In this case the request to `/clientConfig` on the server will have failed, so `theme` will be undefined (it's out of scope in either case). Would it be acceptable to pass in something like `baseThemeName`  in the call to `loadThemedPage()` and use that here? Likewise, I could make that an option in the `secureConfig` collection and refer to that whenever it needs to be used in order to avoid hardcoded strings.

Reply:
If the server doesn't work, would the client be able to operate on its own? If not, it might be more helpful to report an error rather than carrying on the operation.